### ISSUE TITLE
Small example causes require-condition failure in `Clazz.init` 

### ISSUE THREAD

**Issue by fridis** [tokiwa-software/fuzion#979]
```
 > cat test_check_failure2.fz
_ := simpleEfffect.run (unit)->(panic "")

efffect (f ()->unit)
is

  run (_)->unit
  private run(R type, def (unit)->R) R is
    def(unit)

simpleEfffect : efffect ()-> is
 > PRECONDITIONS=true POSTCONDITINS=true ./build/bin/fz test_check_failure2.fz 

error 1: java.lang.Error: require-condition5 failed: dev.flang.air.Clazz:<init>:325
	at dev.flang.util.ANY.require(ANY.java:187)
	at dev.flang.air.Clazz.<init>(Clazz.java:325)
	at dev.flang.air.Clazzes.create(Clazzes.java:367)
	at dev.flang.air.Clazzes.create(Clazzes.java:284)
	at dev.flang.air.Clazz.determineResultClazz(Clazz.java:2193)
	at dev.flang.air.Clazz.dependencies(Clazz.java:411)
	at dev.flang.air.Clazzes.create(Clazzes.java:388)
	at dev.flang.air.Clazzes.clazzWithSpecificOuter(Clazzes.java:1235)
	at dev.flang.air.Clazz.lookup(Clazz.java:1162)
	at dev.flang.air.Clazz.lookup(Clazz.java:1064)
	at dev.flang.air.Clazz.lookup(Clazz.java:1033)
	at dev.flang.air.Clazzes.findClazzes(Clazzes.java:749)
	at dev.flang.air.Clazz$1.action(Clazz.java:1416)
	at dev.flang.ast.Stmnt.visitStatements(Stmnt.java:72)
	at dev.flang.ast.AbstractAssign.visitStatements(AbstractAssign.java:142)
	at dev.flang.ast.AbstractBlock.visitStatements(AbstractBlock.java:101)
	at dev.flang.ast.AbstractBlock.visitStatements(AbstractBlock.java:101)
	at dev.flang.ast.AbstractFeature.visitStatements(AbstractFeature.java:1330)
	at dev.flang.air.Clazz.inspectCode(Clazz.java:1367)
	at dev.flang.air.Clazz.findAllClasses(Clazz.java:1411)
	at dev.flang.air.Clazzes.findAllClasses(Clazzes.java:455)
	at dev.flang.fuir.FUIR.<init>(FUIR.java:196)
	at dev.flang.opt.Optimizer.fuir(Optimizer.java:73)
	at dev.flang.tools.Fuzion$Backend.processFrontEnd(Fuzion.java:367)
	at dev.flang.tools.Fuzion.lambda$parseArgsForBackend$3(Fuzion.java:832)
	at dev.flang.tools.Tool.lambda$run$0(Tool.java:154)
	at dev.flang.util.Errors.runAndExit(Errors.java:704)
	at dev.flang.tools.Tool.run(Tool.java:154)
	at dev.flang.tools.Fuzion.main(Fuzion.java:481)


*** fatal errors encountered, stopping.

```

**Comment by fridis** [comment]
Here is a small example:
```
 > cat test_979a.fz 
xxx := e.r ()->(panic "")
e is
  r ()->unit
  r(R type, def ()->R) R is
    def()
 > PRECONDITIONS=true POSTCONDITINS=true ./build/bin/fz  test_979a.fz 

error 1: java.lang.Error: require-condition5 failed: dev.flang.air.Clazz:<init>:325
	at dev.flang.util.ANY.require(ANY.java:187)
	at dev.flang.air.Clazz.<init>(Clazz.java:325)
	at dev.flang.air.Clazzes.create(Clazzes.java:367)
	at dev.flang.air.Clazzes.create(Clazzes.java:284)
	at dev.flang.air.Clazz.determineResultClazz(Clazz.java:2193)
	at dev.flang.air.Clazz.dependencies(Clazz.java:411)
	at dev.flang.air.Clazzes.create(Clazzes.java:388)
	at dev.flang.air.Clazzes.clazzWithSpecificOuter(Clazzes.java:1235)
	at dev.flang.air.Clazz.lookup(Clazz.java:1162)
	at dev.flang.air.Clazz.lookup(Clazz.java:1064)
	at dev.flang.air.Clazz.lookup(Clazz.java:1033)
	at dev.flang.air.Clazzes.findClazzes(Clazzes.java:749)
	at dev.flang.air.Clazz$1.action(Clazz.java:1416)
	at dev.flang.ast.Stmnt.visitStatements(Stmnt.java:72)
	at dev.flang.ast.AbstractAssign.visitStatements(AbstractAssign.java:142)
	at dev.flang.ast.AbstractBlock.visitStatements(AbstractBlock.java:101)
	at dev.flang.ast.AbstractBlock.visitStatements(AbstractBlock.java:101)
	at dev.flang.ast.AbstractFeature.visitStatements(AbstractFeature.java:1330)
	at dev.flang.air.Clazz.inspectCode(Clazz.java:1367)
	at dev.flang.air.Clazz.findAllClasses(Clazz.java:1411)
	at dev.flang.air.Clazzes.findAllClasses(Clazzes.java:455)
	at dev.flang.fuir.FUIR.<init>(FUIR.java:196)
	at dev.flang.opt.Optimizer.fuir(Optimizer.java:73)
	at dev.flang.tools.Fuzion$Backend.processFrontEnd(Fuzion.java:367)
	at dev.flang.tools.Fuzion.lambda$parseArgsForBackend$3(Fuzion.java:832)
	at dev.flang.tools.Tool.lambda$run$0(Tool.java:154)
	at dev.flang.util.Errors.runAndExit(Errors.java:704)
	at dev.flang.tools.Tool.run(Tool.java:154)
	at dev.flang.tools.Fuzion.main(Fuzion.java:481)


*** fatal errors encountered, stopping.

```

**Comment by fridis** [comment]
The fix #982 results in the first example failing like in #978 now, so this issue can be closed even though the test does not work yet.
